mautrix_instagram/portal.py

```diff
@@ -972,7 +971,21 @@ async def _handle_instagram_media_share(
             format=Format.HTML,
             external_url=external_url,
         )
-        await self._send_message(intent, caption, timestamp=item.timestamp_ms)
+
+        if self.bridge.config["bridge.caption_in_message"]:
+            combined = content
+            combined["filename"] = combined.body
+            caption_body = f"{prefix.body}\n{caption.body}"
+            combined["org.matrix.msc1767.text"] = caption_body
+            combined.body = caption_body
+            combined["format"] = str(Format.HTML)
+            combined["formatted_body"] = f"{prefix.formatted_body}<br />{caption.formatted_body}"
```

Comment:
This should probably also handle errors where the returned content will also be text (so merge prefix+content+caption into the resulting body and don't set filename)